### PR TITLE
Prevent unneeded get_data_from_viewer call on app startup

### PR DESCRIPTION
**Pull request by rosteen**
This code block was getting triggered on app startup while the traitlet was still the default empty string rather than the assigned default value. This is a negligible performance increase but avoiding this block on startup matters for some other work I'm doing.

### CODE REVIEW

**Review by kecnry — APPROVED** [review]
oh it was triggering on the empty string?  Do we know how this happened after the data was already loaded but before the options for the dropdown were set? 😬 

Code changes look reasonable enough though (I'm just wondering if we have a more-general problem we should try to address lower in the logic).  `in` might be slightly easier to read, but that might be personal preference.

**Comment by rosteen** [comment]
This was happening before data load, on app initialization.

File: setup.cfg [hunk 1]
```diff
@@ -103,6 +103,7 @@ filterwarnings =
     ignore:.*np\.int0.*is a deprecated alias for:DeprecationWarning
     ignore:zmq\.eventloop\.ioloop is deprecated in pyzmq:DeprecationWarning
     ignore:AsdfInFits has been deprecated
+    ignore:specutils uses the deprecated entry point asdf_extensions
```
Comment (pllim) [inline setup.cfg:106]: This should not have been here. How did this sneak in?

Comment (rosteen) [reply setup.cfg:106]: I have absolutely no idea 🤔

Comment (rosteen) [reply setup.cfg]: Got rid of it with a rebase 🤷

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**